### PR TITLE
[AS7326-56X] Support mulit PSU SN in PDDF

### PR DESCRIPTION
**Pull request by jostar-yang**
Signed-off-by: Jostar Yang <jostar_yang@accton.com.tw>

#### Why I did it
AS7326  support 3Y and ACBEL PSU. So it need customer drv to handle access psu serial_number and model_name on different PSU.
#### How I did it
Implement pddf_custom_psu.c drv code.
#### How to verify it
#pddf_psuutil mfrinfo

PSU    Status    Manufacturer ID    Model       Serial            Fan Airflow Direction
-----  --------  -----------------  ----------  ----------------  -----------------------
PSU1   OK        ACBEL              FSF019-610  FSF0191813105520  EXHAUST
PSU2   OK        ACBEL              FSF019-610  FSF0191813105521  EXHAUST

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    struct pddf_psu_data data;
+    int i, status;
+
+    for (i = 0; i < ARRAY_SIZE(models); i++) {
+        memset(data.serial_number, 0, sizeof(data.serial_number));
+
+        status = pddf_psu_read_block(client, models[i].offset,
+                                           data.model_name, models[i].length);
+        if (status < 0) {
+            data.serial_number[0] = '\0';
```
[the remark this answers is not in the document]
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c]: Ok. I update it.

File: platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            dev_dbg(&client->dev, "unable to read model name from (0x%x) offset(0x%x)\n",
+                                  client->addr, serials[i].offset);
+            return status;
+        }
+        else {
+            data.serial_number[serials[i].length] = '\0';
```
[the remark this answers is not in the document]
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c]: Ok. I updated it.

**Review by FuzailBrcm — APPROVED** [review]

File: platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+struct model_name_info models[] = {
+{PSU_TYPE_AC_110V, 0x20, 8, 8,  "YM-2651Y"},
+{PSU_TYPE_DC_48V,  0x20, 8, 8,  "YM-2651V"},
+{PSU_TYPE_DC_12V,  0x00, 11, 11, "PSU-12V-750"},
+{PSU_TYPE_AC_ACBEL_FSF019, 0x15, 10, 7, "FSF019-"}
+
+};
+
+struct serial_number_info serials[] = {
+{PSU_TYPE_AC_110V, 0x2e, 18, 18,  "YM-2651Y"},
```
Comment (FuzailBrcm) [inline platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c:54]: Can you update this and change the offset if required?

Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/pddf_custom_psu.c]: I updated it. Please check.

**Review by FuzailBrcm — APPROVED** [review]
Changes are fine

File: platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/Makefile [hunk 4]
```diff
@@ -1,7 +1,12 @@
 ifneq ($(KERNELRELEASE),)
 obj-m:= accton_i2c_cpld.o \
     accton_as7326_56x_fan.o accton_as7326_56x_leds.o \
-    accton_as7326_56x_psu.o ym2651y.o
+    accton_as7326_56x_psu.o ym2651y.o \
+    pddf_custom_psu.o
+
+	    
+CFLAGS_pddf_custom_psu.o := -I$(M)/../../../../pddf/i2c/modules/include
+KBUILD_EXTRA_SYMBOLS := $(M)/../../../../pddf/i2c/Module.symvers.PDDF
```
Comment (FuzailBrcm) [inline platform/broadcom/sonic-platform-modules-accton/as7326-56x/modules/Makefile:9]: This change requires the changes from https://github.com/Azure/sonic-buildimage/pull/9582

**Review by prgeor — DISMISSED** [review]

**Review by FuzailBrcm — APPROVED** [review]
Changes look good to me. 
However, this change require the following PR to be merged 
[AS7326-5X] Fix code bug for led drv (https://github.com/Azure/sonic-buildimage/pull/8555)

**Comment by jostar-yang** [comment]
Close this PR because PR10053 remove pddf files. It cause file conflict, but no write access here. This PR become invalid. So close and send as7326 needed file in PR https://github.com/Azure/sonic-buildimage/pull/10340.

**Pull request closed without merging**